runtime/lua/vim/filetype.lua

```diff
@@ -2267,11 +2267,6 @@ local pattern = {
     ['/%.init/.*%.conf$'] = 'upstart',
     ['/xorg%.conf%.d/.*%.conf$'] = detect.xfree86_v4,
   },
-  ['sst%.meta'] = {
```

Comment:
Similar to `'calendar/'` and `'cmus/'` case but not exactly. I would expect that most of inputs don't match both `'sst%.meta'` and `'%.[-_]?sst%.meta$'` patterns, but it is faster to try to match the one without quantifiers. And either of them *should* be matched for every input pattern, so this matters.

Although the slowdown for `'%.[-_]?sst%.meta$'` is small enough to maybe justify moving three separate patterns into one.

What *should* have been done better is using anchored parent pattern: `'sst%.meta$'` . I'd personally just change this, but have no objections to replacing three patterns with one if this makes maintenance easier.